distribution/lib/Standard/AWS/0.0.0-dev/src/S3/S3_File.enso

```diff
@@ -185,22 +189,36 @@ type S3_File
        - destination: the destination to move the file to.
        - replace_existing: specifies if the operation should proceed if the
          destination file already exists. Defaults to `False`.
-    copy_to : S3_File -> Boolean -> Any ! File_Error
-    copy_to self (destination : Writable_File) replace_existing=False =
+    copy_to : Writable_File -> Boolean -> Any ! File_Error
+    copy_to self (destination : Writable_File) (replace_existing : Boolean = False) =
         if self.is_directory then Error.throw (S3_Error.Error "Copying S3 folders is currently not implemented." self.uri) else
             if Context.Output.is_enabled.not then Error.throw (Forbidden_Operation.Error "Copying an S3_File is forbidden as the Output context is disabled.") else
                 case destination.file of
                     # Special shortcut for more efficient handling of S3 file copying (no need to move the data to our machine)
                     s3_destination : S3_File ->
                         if replace_existing.not && s3_destination.exists then Error.throw (File_Error.Already_Exists destination) else
                             translate_file_errors self <| S3.copy_object self.bucket self.prefix s3_destination.bucket s3_destination.prefix self.credentials . if_not_error <| s3_destination
+                    _ -> generic_copy self destination.file replace_existing
+
+    ## Moves the file to the specified destination.
+
+       ! S3 Move is a Copy and Delete
+
+         Since S3 does not support moving files, this operation is implemented
+         as a copy followed by delete. Keep in mind that the space usage of the
+         file will briefly be doubled and that the operation may not be as fast
+         as a local move often is.
 
-                    # Generic implementation using streams
-                    _ ->
-                        self.with_input_stream [File_Access.Read] input_stream->
-                            open_settings = if replace_existing then [File_Access.Write, File_Access.Create, File_Access.Truncate_Existing] else [File_Access.Write, File_Access.Create_New]
-                            destination.with_output_stream open_settings output_stream->
-                                output_stream.write_stream input_stream
+       Arguments:
+       - destination: the destination to move the file to.
+       - replace_existing: specifies if the operation should proceed if the
+         destination file already exists. Defaults to `False`.
+    move_to : Writable_File -> Boolean -> Nothing ! File_Error
+    move_to self (destination : Writable_File) (replace_existing : Boolean = False) =
+        if Context.Output.is_enabled.not then Error.throw (Forbidden_Operation.Error "File moving is forbidden as the Output context is disabled.") else
```

Comment:
Not for this one, but I feel like we need a better construct for this. Something like

```suggestion
        Context.Output.when_enabled disabled_message="File moving is forbidden as the Output context is disabled.") <|
```

Reply:
Good idea, I created a ticket so that we can pick it up whenever we can: #9080